src/librustc_mir/interpret/memory.rs

```diff
@@ -535,6 +535,19 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         id: AllocId,
         liveness: AllocCheck,
     ) -> InterpResult<'static, (Size, Align)> {
+        // Allocations of `static` items
+        // Can't do this in the match argument, we may get cycle errors since the lock would
+        // be held throughout the match.
+        let alloc = self.tcx.alloc_map.lock().get(id);
```

Comment:
Isn't it very expensive to always try to get this global lock?

Also a comment should explain why this is done first.

Reply:
Maybe a better ordering here is to first lookup directly in `alloc_map` (that will cover the vast majority of cases), then ask the global map, and only then proceed with `get`.